src/main/java/seedu/address/commons/util/DateUtil.java

```diff
@@ -1,61 +1,96 @@
 package seedu.address.commons.util;
 
 import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.core.Messages.MESSAGE_PARSER_DATE_CONSTRAINTS;
 
 import java.time.LocalDate;
 import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeFormatterBuilder;
 import java.time.format.DateTimeParseException;
+import java.time.format.ResolverStyle;
 
 import seedu.address.commons.exceptions.DateConversionException;
 
 /**
  * A class for encoding and decoding of Dates.
  */
 public class DateUtil {
+    /*
+        Inconsistency in naming of year: uuuu in formatters, yyyy in error messages.
+        Java uses uuuu for years (yyyy is year-of-era).
+        However, yyyy is the more commonly accepted symbol for year and will be more intuitive for users.
+     */
+    private static DateTimeFormatter dateFormat1 = DateTimeFormatter.ofPattern("dd-MM-uuuu");
+    private static DateTimeFormatter dateFormat2 = DateTimeFormatter.ofPattern("ddMMuuuu");
+    private static DateTimeFormatter dateFormat3 = DateTimeFormatter.ofPattern("dd/MM/uuuu");
+    private static DateTimeFormatter dateformat4 = DateTimeFormatter.ofPattern("dd.MM.uuuu");
+
+    private static DateTimeFormatter formatter =
+            new DateTimeFormatterBuilder()
+                    .appendOptional(dateFormat1)
+                    .appendOptional(dateFormat2)
+                    .appendOptional(dateFormat3)
+                    .appendOptional(dateformat4)
+                    .toFormatter()
+                    .withResolverStyle(ResolverStyle.STRICT);
 
     /**
      * Encodes a date passed as a String into a LocalDate.
-     * @param date Date in the yyyy-mm-dd format.
+     * @param date Date in the uuuu-mm-dd format.
      * @return A LocalDate object.
      * @throws DateConversionException Occurs when a date had been passed in with the wrong format.
      */
     public static LocalDate encodeDate(String date) throws DateConversionException {
         requireNonNull(date);
         try {
-            return LocalDate.parse(date, DateTimeFormatter.ofPattern("dd-MM-yyyy"));
+            return checkDateIsNotNegative(LocalDate.parse(date, formatter));
         } catch (DateTimeParseException e) {
```

Comment:
Does it need to catch DateConversionException too, given that it is thrown in checkDateIsNotNegative? Or is DateConversionException a subclass of DateTimeParseException?

Reply:
I checked and it doesn't appear to be a subclass of DateTimeParseException

Reply:
This is handled by throws statement in the method header. Would it be clearer if i explicitly caught and then threw it again?

Reply:
Hmm, because they throw with different messages, perhaps can leave it as it is?